controller/src/main/java/io/pravega/controller/metrics/StreamMetrics.java

```diff
@@ -57,10 +76,11 @@
      */
     public void createStream(String scope, String streamName, int minNumSegments, Duration latency) {
         DYNAMIC_LOGGER.incCounterValue(CREATE_STREAM, 1);
-        DYNAMIC_LOGGER.reportGaugeValue(nameFromStream(OPEN_TRANSACTIONS, scope, streamName), 0);
+        // We need to initialize the number of initial segments, but prevent this process to continue reporting the
+        // segment count for this Stream. This will avoid multiple controllers to report conflicting values.
         DYNAMIC_LOGGER.reportGaugeValue(nameFromStream(SEGMENTS_COUNT, scope, streamName), minNumSegments);
-        DYNAMIC_LOGGER.incCounterValue(nameFromStream(SEGMENTS_SPLITS, scope, streamName), 0);
-        DYNAMIC_LOGGER.incCounterValue(nameFromStream(SEGMENTS_MERGES, scope, streamName), 0);
+        executor.schedule(() -> DYNAMIC_LOGGER.freezeGaugeValue(nameFromStream(SEGMENTS_COUNT, scope, streamName)),
```

Comment:
You run this every few seconds. If you don't create the thread pool using `ExecutorServiceHelpers`, this task will continue running after you invoke `shutdown`.